qutip/qip/device/processor.py

```diff
@@ -533,16 +532,22 @@ def get_qobjevo(self, args=None, noisy=False):
 
         qu_list = []
         c_ops = []
+        t_lists = []
```

Comment:
As a QuTiP user, I would be pretty horrified to discover that `self.tlist` was set by calling `.get_qobjevo(...)` so let's find another way to do this? E.g. just add a helper method that returns the full tlist and call that alongside `.get_objevo` before calling the solver?

The original code was definitely odd here -- merging the tlists in all the coefficients didn't make a lot of sense.

Reply:
I made a `tlist` method. Adding `QobjEvo` with different `tlist` is not allowed in v4, thus the `merge_qobjevo`, but I don't understand why the solver `tlist` is the merged pulse `tlist`.

Reply:
The solver `tlist` being the merged pulse `tlist` is odd.

I'm guessing it's a heuristic so that the user doesn't need to supply one, but I'm not sure it's a good heuristic. Not all QobjEvos even have tlists.

A better heuristic would maybe be a tlist that includes the start and end of each pulse, any tlist values from each pulse (for step function pulses that use tlists) and enough times in between those to make the solver happy.

@BoxiLi Do you know why `.run_state` uses this merged tlist?

I'm happy for us to keep the old behaviour in this PR -- hopefully @BoxiLi can open an appropriate issue for us if the tlist needs to change in v5.

Reply:
Leaving this unresolved so that @BoxiLi can see it easily, but we can consider it resolved as far as this PR goes.